shared/key_vault_secret.py

```diff
@@ -5,6 +5,7 @@
 
 def get_key_vault_secret(azure_credential, key_vault_name, secret_name):
 
+    # TODO: Does this need to have hyphens stripped?

```

Comment:
I put this TODO here as I'm wondering if others will run into the hyphen problem that I ran into. I use hyphens all the time in my resource names.